rebase_v1.13.1_report.md

```diff
@@ -0,0 +1,38 @@
+# Rebase to v1.13.1 - Anticipated Challenges
+
+## Snapshot of Divergence
+- Local branch `main` (origin/main) descends from `v1.11.3`; merge-base with `v1.13.1` is commit `a7ed34658`.
+- `origin/main` is **270 commits** ahead of `v1.11.3` and **501 commits** behind `v1.13.1`; there are 82 merge commits and 123 commits containing `UPSTREAM` markers that will need manual triage.
+- Numerous cherry-picked upstream fixes (e.g., PRs #6692, #6836, #7083) already exist in the target release, implying they must be dropped or squashed during the rebase.
+
+## Carry Patches and OpenShift Packaging
+- Long-running OpenShift carry chain touches `.ci-operator.yaml`, `Dockerfile.ocp`, `Dockerfile.openshift`, `Makefile` overrides (`test`, `dep-ensure`), and repeated image metadata synchronizations with ART. These files do not exist upstream; they must be reintroduced cleanly after the rebase.
+- The external plugin `ocp_dnsnameresolver` is currently vendored along with large slices of `github.com/openshift/{api,client-go,...}`. Upstream `v1.13.1` has no `vendor/` tree, so we must decide between re-vendoring the entire OpenShift dependency set or switching to module-aware builds that can fetch these deps.
+- Several carry commits revert or modify upstream behavior (e.g., wildcard warning toggles, cache CD-bit handling, bufsize defaults). Each needs reassessment against upstream changes to avoid reintroducing bugs that have since been fixed differently.
+
+## Toolchain and Dependency Shifts
+- Upstream now enforces Go **1.24+** (release notes call out 1.25.2) via `.go-version` and `GOTOOLCHAIN`. Our builders and CI images must ship that compiler, otherwise `go` will attempt to auto-download toolchains (blocked in disconnected builds).
+- `go.mod` migrates from legacy dependencies (`aws-sdk-go` v1, DataDog v1, etc.) to new major versions (`aws-sdk-go-v2`, `dd-trace-go/v2`, OpenTelemetry packages, etc.), and bumps Kubernetes libs to **v0.34.1**. These introduce breaking API changes that our OpenShift carry code (including the external plugin) must accommodate.
```

Comment:
> `go.mod` migrates from legacy dependencies (`aws-sdk-go` v1, DataDog v1, etc.) to new major versions (`aws-sdk-go-v2`, `dd-trace-go/v2`, OpenTelemetry packages, etc.)

Do we enable any plugins that use these packages?  

> These introduce breaking API changes that our OpenShift carry code (including the external plugin) must accommodate.

Seems we need to rebase https://github.com/openshift/coredns-ocp-dnsnameresolver then too.

Reply:
revised the rebase report to highlight plugin-level changes, will link the section after next push